src/internal/plugins/controllers/drawers.ts

```diff
@@ -17,30 +24,39 @@ export interface DrawerConfig {
   onResize?: NonCancelableEventHandler<{ size: number; id: string }>;
   orderPriority?: number;
   defaultActive?: boolean;
-  trigger: {
+  trigger?: {
     iconSvg: string;
   };
-  mountContent: (container: HTMLElement) => void;
+  mountContent: (container: HTMLElement, mountContext?: MountContentContext) => void;
```

Comment:
**Before merge:**

How about making `mountContext` always defined and provide a no-op `onVisibilityChange` for unsupported case, but allow teams to avoid unnecessary undefined checks?

Reply:
done
btw, there is no even need to define an noop function, since `onVisibilityChange` is always defined: https://github.com/cloudscape-design/components/blob/feat/app-layout-global-drawers/src/internal/plugins/helpers/runtime-content-wrapper.tsx#L25